closed/src/java.base/share/classes/sun/security/ec/NativeECDHKeyAgreement.java

```diff
@@ -198,8 +209,10 @@ protected Key engineDoPhase(Key key, boolean lastPhase)
                 ("Key must be an instance of PublicKey");
         }
 
-        // Validate public key.
-        validate(privateKeyOps, ecKey);
+        // Validate public key when we are not making use of a brainpoolP512r1 based key.
+        if (!(ECUtil.isBrainpoolP512r1(this.privateKey.getParams()))) {
```

Comment:
Suggest removing the unnecessary parentheses (this occurs multiple places)
```java
        if (!NativeECUtil.isBrainpoolP512r1(this.privateKey.getParams())) {
```
The new method should be added to `NativeECUtil`, not `ECUtil`.

Reply:
The method `isBrainpoolP512r1` was moved to the `NativeECUtil` class. Redundant parentheses were removed here.